src/main/java/com/metamx/common/CompressionUtils.java

```diff
@@ -0,0 +1,348 @@
+   */
+  public static GZIPInputStream gzipInputStream(final InputStream in) throws IOException
+  {
+    return new GZIPInputStream(
+        new FilterInputStream(in)
+        {
+          @Override
+          public int available()
+          {
+            return 1 << 10; // http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7036144
+          }
+
+        }
+    );
+  }
+
+  public static long gunzip(InputStream in, OutputStream out) throws IOException
+  {
+    try (GZIPInputStream gzipInputStream = gzipInputStream(in)) {
+      return ByteStreams.copy(gzipInputStream, out);
+    }
+    finally {
+      out.close();
+    }
+  }
+
+  /**
+   * A gunzip function to store locally
+   *
+   * @param in          The factory to produce input streams
+   * @param outFile     The file to store the result into
+   * @param shouldRetry A predicate to indicate if the Throwable is recoverable
+   *
+   * @return The count of bytes written to outFile
+   */
+  public static FileUtils.FileCopyResult gunzip(
+      final ByteSource in,
+      final File outFile,
+      Predicate<Throwable> shouldRetry
+  )
+  {
+    return FileUtils.retryCopy(
+        new ByteSource()
+        {
+          @Override
+          public InputStream openStream() throws IOException
+          {
+            return gzipInputStream(in.openStream());
+          }
+        },
+        shouldRetry,
+        outFile,
+        10
+    );
+  }
+
+
+  public static FileUtils.FileCopyResult gunzip(final ByteSource in, File outFile)
+  {
+    return gunzip(in, outFile, FileUtils.MD5ERROR);
+  }
+
+  public static long gzip(InputStream inputStream, OutputStream out) throws IOException
+  {
+    try (GZIPOutputStream outputStream = new GZIPOutputStream(out)) {
+      return ByteStreams.copy(inputStream, outputStream);
+    }
+    finally {
+      out.close();
+      inputStream.close();
+    }
+  }
+
+  public static FileUtils.FileCopyResult gzip(final File inFile, final File outDir, Predicate<Throwable> shouldRetry)
+      throws IOException
+  {
+    final File outFile = Paths.get(outDir.getAbsolutePath(), inFile.getName() + GZ_SUFFIX).toFile();
+    FileUtils.retryCopy(
+        new ByteSource()
+        {
+          @Override
+          public InputStream openStream() throws IOException
+          {
+            return new FileInputStream(inFile);
+          }
+        },
+        shouldRetry,
+        new ByteSink()
+        {
+          @Override
+          public OutputStream openStream() throws IOException
+          {
+            return new GZIPOutputStream(new FileOutputStream(outFile));
+          }
+        },
+        10
+    );
+    return FileUtils.FileCopyResult.buildResult(outFile.toPath(), "MD5");
+  }
+
+
+  public static FileUtils.FileCopyResult gzip(final File inFile, final File outDir) throws IOException
+  {
+    return gzip(inFile, outDir, FileUtils.MD5ERROR);
+  }
+
+  public static Boolean isZip(String fName)
```

Comment:
It is public because if a file name returns true from that function, then the library should be able to use it just fine. 

I had a lot of checks in various functions that had pretty much this exact same function. So instead of having it copied everywhere I have them using this method instead.